File: Sovenire_Collenction_Backend/Models/Review.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Souvenir_Collection_Backend.Models
{
    public class Review
    {
        [Key]
        public Guid Id { get; set; } = Guid.NewGuid();

        public Guid UserId { get; set; }

        [ForeignKey(nameof(UserId))]
        public User? User { get; set; }

        public Guid? ProductId { get; set; }

        [ForeignKey(nameof(ProductId))]
        public Product? Product { get; set; }

        public Guid? CollectionId { get; set; }

        [ForeignKey(nameof(CollectionId))]
        public Collection? Collection { get; set; }

        public string? ReviewText { get; set; }    

        [Range(1, 5)]
        public int Rating { get; set; } = 1;        

        public string Image { get; set; } = string.Empty;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: Souvenir_Collection_Frontend/lib/models/review.dart
class Review {
  final String id;
  final String userId;
  final String? productId;
  final String? reviewText;
  final int rating;
  final String image;
  final DateTime? createdAt;

  const Review({
    required this.id,
    required this.userId,
    this.productId,
    this.reviewText,
    this.rating = 0,
    this.image = '',
    this.createdAt,
  });

  factory Review.fromJson(Map<String, dynamic> json) {
    return Review(
      id: json['id'] as String? ?? '',
      userId: json['userId'] as String? ?? '',
      productId: json['productId'] as String?,
      reviewText: json['reviewText'] as String?,
      rating: json['rating'] as int? ?? 0,
      image: json['image'] as String? ?? '',
      createdAt: json['createdAt'] != null
          ? DateTime.tryParse(json['createdAt'] as String)
          : null,
    );
  }

  Map<String, dynamic> toJson() => {
        'id': id,
        'userId': userId,
        'productId': productId,
        'reviewText': reviewText,
        'rating': rating,
        'image': image,
        'createdAt': createdAt?.toIso8601String(),
      };
}
